examples/unsloth/unsloth.yaml

```diff
@@ -0,0 +1,31 @@
+# Runs the unsloth example app via SkyPilot
```

Comment:
I ran this and the job completed successfully, but my bucket is still empty..

```
$ ssh myclus ls -lah /outputs
Warning: Permanently added '34.134.188.34' (ED25519) to the list of known hosts.
total 0
```

Logs:
```
(task, pid=12349) ==((====))==  Unsloth: Fast Mistral patching release 2024.2
(task, pid=12349)    \\   /|    GPU: Tesla T4. Max memory: 14.581 GB. Platform = Linux.
(task, pid=12349) O^O/ \_/ \    Pytorch: 2.2.0+cu118. CUDA = 7.5. CUDA Toolkit = 11.8.
(task, pid=12349) \        /    Bfloat16 = FALSE. Xformers = 0.0.24+cu118. FA = False.
(task, pid=12349)  "-____-"     Free Apache license: http://github.com/unslothai/unsloth
model.safetensors: 100%|██████████| 4.13G/4.13G [00:20<00:00, 201MB/s]20<00:00, 206MB/s] 
generation_config.json: 100%|██████████| 116/116 [00:00<00:00, 666kB/s]
tokenizer_config.json: 100%|██████████| 971/971 [00:00<00:00, 6.18MB/s]
tokenizer.model: 100%|██████████| 493k/493k [00:00<00:00, 186MB/s]
tokenizer.json: 100%|██████████| 1.80M/1.80M [00:00<00:00, 30.1MB/s]
special_tokens_map.json: 100%|██████████| 438/438 [00:00<00:00, 3.34MB/s]
(task, pid=12349) Unsloth 2024.2 patched 32 layers with 32 QKV layers, 32 O layers and 32 MLP layers.
Map: 100%|██████████| 210289/210289 [00:27<00:00, 7752.18 examples/s]398.76 examples/s]
(task, pid=12349) ==((====))==  Unsloth - 2x faster free finetuning | Num GPUs = 1
(task, pid=12349)    \\   /|    Num examples = 210,289 | Num Epochs = 1
(task, pid=12349) O^O/ \_/ \    Batch size per device = 2 | Gradient Accumulation steps = 4
(task, pid=12349) \        /    Total batch size = 8 | Total steps = 60
(task, pid=12349)  "-____-"     Number of trainable parameters = 41,943,040
{'loss': 2.2025, 'grad_norm': 2.0635592937469482, 'learning_rate': 5e-06, 'epoch': 0.0}
{'loss': 1.8396, 'grad_norm': 2.085563898086548, 'learning_rate': 1e-05, 'epoch': 0.0}
{'loss': 2.2164, 'grad_norm': 2.3598461151123047, 'learning_rate': 1.5e-05, 'epoch': 0.0}
{'loss': 1.848, 'grad_norm': 3.40551495552063, 'learning_rate': 2e-05, 'epoch': 0.0}
{'loss': 2.1649, 'grad_norm': 2.8964200019836426, 'learning_rate': 2.5e-05, 'epoch': 0.0}
{'loss': 2.0281, 'grad_norm': 2.397609233856201, 'learning_rate': 3e-05, 'epoch': 0.0}
{'loss': 2.1216, 'grad_norm': 1.212262511253357, 'learning_rate': 3.5e-05, 'epoch': 0.0}
{'loss': 1.5452, 'grad_norm': 1.369713544845581, 'learning_rate': 4e-05, 'epoch': 0.0}
{'loss': 1.5235, 'grad_norm': 1.4552085399627686, 'learning_rate': 4.5e-05, 'epoch': 0.0}
{'loss': 1.3379, 'grad_norm': 1.1726322174072266, 'learning_rate': 5e-05, 'epoch': 0.0}
{'loss': 2.0901, 'grad_norm': 1.2783335447311401, 'learning_rate': 4.9e-05, 'epoch': 0.0}
{'loss': 1.5278, 'grad_norm': 1.591582179069519, 'learning_rate': 4.8e-05, 'epoch': 0.0}
{'loss': 1.3301, 'grad_norm': 1.5074691772460938, 'learning_rate': 4.7e-05, 'epoch': 0.0}
{'loss': 1.5371, 'grad_norm': 1.1644656658172607, 'learning_rate': 4.600000000000001e-05, 'epoch': 0.0}
{'loss': 1.3483, 'grad_norm': 1.5559577941894531, 'learning_rate': 4.5e-05, 'epoch': 0.0}
(task, pid=12349) {'loss': 1.4178, 'grad_norm': 1.4345111846923828, 'learning_rate': 4.4000000000000006e-05, 'epoch': 0.0}
{'loss': 1.2986, 'grad_norm': 1.4233272075653076, 'learning_rate': 4.3e-05, 'epoch': 0.0}
{'loss': 1.4794, 'grad_norm': 1.234622836112976, 'learning_rate': 4.2e-05, 'epoch': 0.0}
{'loss': 1.6199, 'grad_norm': 1.1267141103744507, 'learning_rate': 4.1e-05, 'epoch': 0.0}
{'loss': 1.4525, 'grad_norm': 1.413191318511963, 'learning_rate': 4e-05, 'epoch': 0.0}
{'loss': 1.4184, 'grad_norm': 1.5253208875656128, 'learning_rate': 3.9000000000000006e-05, 'epoch': 0.0}
{'loss': 1.6283, 'grad_norm': 1.2969868183135986, 'learning_rate': 3.8e-05, 'epoch': 0.0}
{'loss': 1.4427, 'grad_norm': 1.1964991092681885, 'learning_rate': 3.7e-05, 'epoch': 0.0}
{'loss': 1.3099, 'grad_norm': 1.1834884881973267, 'learning_rate': 3.6e-05, 'epoch': 0.0}
{'loss': 1.1245, 'grad_norm': 1.3284245729446411, 'learning_rate': 3.5e-05, 'epoch': 0.0}
{'loss': 1.2279, 'grad_norm': 1.0646255016326904, 'learning_rate': 3.4000000000000007e-05, 'epoch': 0.0}
{'loss': 1.3238, 'grad_norm': 1.2521451711654663, 'learning_rate': 3.3e-05, 'epoch': 0.0}
{'loss': 1.8785, 'grad_norm': 1.6011723279953003, 'learning_rate': 3.2000000000000005e-05, 'epoch': 0.0}
{'loss': 1.3141, 'grad_norm': 1.4861927032470703, 'learning_rate': 3.1e-05, 'epoch': 0.0}
{'loss': 1.0967, 'grad_norm': 1.1702327728271484, 'learning_rate': 3e-05, 'epoch': 0.0}
{'loss': 1.498, 'grad_norm': 1.2589576244354248, 'learning_rate': 2.9e-05, 'epoch': 0.0}
{'loss': 1.2282, 'grad_norm': 1.0961182117462158, 'learning_rate': 2.8000000000000003e-05, 'epoch': 0.0}
{'loss': 1.1004, 'grad_norm': 1.1435779333114624, 'learning_rate': 2.7000000000000002e-05, 'epoch': 0.0}
{'loss': 1.6536, 'grad_norm': 1.1448558568954468, 'learning_rate': 2.6000000000000002e-05, 'epoch': 0.0}
{'loss': 1.7333, 'grad_norm': 1.120127558708191, 'learning_rate': 2.5e-05, 'epoch': 0.0}
{'loss': 1.0804, 'grad_norm': 1.1107993125915527, 'learning_rate': 2.4e-05, 'epoch': 0.0}
{'loss': 1.1516, 'grad_norm': 0.9329462051391602, 'learning_rate': 2.3000000000000003e-05, 'epoch': 0.0}
{'loss': 1.1166, 'grad_norm': 1.119095802307129, 'learning_rate': 2.2000000000000003e-05, 'epoch': 0.0}
{'loss': 1.5899, 'grad_norm': 1.3269765377044678, 'learning_rate': 2.1e-05, 'epoch': 0.0}
{'loss': 1.6088, 'grad_norm': 1.2719619274139404, 'learning_rate': 2e-05, 'epoch': 0.0}
{'loss': 1.0895, 'grad_norm': 1.2886533737182617, 'learning_rate': 1.9e-05, 'epoch': 0.0}
{'loss': 1.5897, 'grad_norm': 1.148683786392212, 'learning_rate': 1.8e-05, 'epoch': 0.0}
{'loss': 1.4424, 'grad_norm': 1.3090258836746216, 'learning_rate': 1.7000000000000003e-05, 'epoch': 0.0}
{'loss': 1.1811, 'grad_norm': 1.3880740404129028, 'learning_rate': 1.6000000000000003e-05, 'epoch': 0.0}
{'loss': 1.1768, 'grad_norm': 1.132697582244873, 'learning_rate': 1.5e-05, 'epoch': 0.0}
{'loss': 1.6898, 'grad_norm': 1.2975410223007202, 'learning_rate': 1.4000000000000001e-05, 'epoch': 0.0}
{'loss': 1.2263, 'grad_norm': 1.163159728050232, 'learning_rate': 1.3000000000000001e-05, 'epoch': 0.0}
{'loss': 1.3278, 'grad_norm': 1.0435025691986084, 'learning_rate': 1.2e-05, 'epoch': 0.0}
{'loss': 1.3001, 'grad_norm': 1.0292456150054932, 'learning_rate': 1.1000000000000001e-05, 'epoch': 0.0}
{'loss': 1.4109, 'grad_norm': 1.0305780172348022, 'learning_rate': 1e-05, 'epoch': 0.0}
{'loss': 1.5118, 'grad_norm': 1.0234051942825317, 'learning_rate': 9e-06, 'epoch': 0.0}
{'loss': 1.5512, 'grad_norm': 1.0321251153945923, 'learning_rate': 8.000000000000001e-06, 'epoch': 0.0}
{'loss': 1.3609, 'grad_norm': 1.0848301649093628, 'learning_rate': 7.000000000000001e-06, 'epoch': 0.0}
{'loss': 1.449, 'grad_norm': 1.1692029237747192, 'learning_rate': 6e-06, 'epoch': 0.0}
{'loss': 1.3583, 'grad_norm': 1.129260540008545, 'learning_rate': 5e-06, 'epoch': 0.0}
{'loss': 1.2245, 'grad_norm': 4.09239387512207, 'learning_rate': 4.000000000000001e-06, 'epoch': 0.0}
{'loss': 1.4048, 'grad_norm': 1.2658228874206543, 'learning_rate': 3e-06, 'epoch': 0.0}
{'loss': 1.265, 'grad_norm': 1.2331774234771729, 'learning_rate': 2.0000000000000003e-06, 'epoch': 0.0}
{'loss': 1.0502, 'grad_norm': 1.0950510501861572, 'learning_rate': 1.0000000000000002e-06, 'epoch': 0.0}
{'loss': 1.3244, 'grad_norm': 1.391110897064209, 'learning_rate': 0.0, 'epoch': 0.0}
{'train_runtime': 294.2327, 'train_samples_per_second': 1.631, 'train_steps_per_second': 0.204, 'train_loss': 1.473148876428604, 'epoch': 0.0}
100%|██████████| 60/60 [04:54<00:00,  4.90s/it]
INFO: Job finished (status: SUCCEEDED).
Shared connection to 34.134.188.34 closed.
I 02-26 10:01:32 cloud_vm_ray_backend.py:3212] Job ID: 1
I 02-26 10:01:32 cloud_vm_ray_backend.py:3212] To cancel the job:       sky cancel myclus 1
I 02-26 10:01:32 cloud_vm_ray_backend.py:3212] To stream job logs:      sky logs myclus 1
I 02-26 10:01:32 cloud_vm_ray_backend.py:3212] To view the job queue:   sky queue myclus
I 02-26 10:01:32 cloud_vm_ray_backend.py:3308] 
I 02-26 10:01:32 cloud_vm_ray_backend.py:3308] Cluster name: myclus
I 02-26 10:01:32 cloud_vm_ray_backend.py:3308] To log into the head VM: ssh myclus
I 02-26 10:01:32 cloud_vm_ray_backend.py:3308] To submit a job:         sky exec myclus yaml_file
I 02-26 10:01:32 cloud_vm_ray_backend.py:3308] To stop the cluster:     sky stop myclus
I 02-26 10:01:32 cloud_vm_ray_backend.py:3308] To teardown the cluster: sky down myclus
D 02-26 10:01:32 skypilot_config.py:136] Using config path: /Users/romilb/.sky/config.yaml
D 02-26 10:01:32 skypilot_config.py:140] Config loaded:
D 02-26 10:01:32 skypilot_config.py:140] {'kubernetes': {'ports': 'ingress'},
D 02-26 10:01:32 skypilot_config.py:140]  'serve': {'controller': {'resources': {'cloud': 'kubernetes'}}}}
D 02-26 10:01:32 skypilot_config.py:150] Config syntax check passed.
Clusters
NAME    LAUNCHED    RESOURCES                                       STATUS  AUTOSTOP  COMMAND                       
myclus  8 mins ago  1x GCP(n1-highmem-4, {'T4': 1}, disk_size=128)  UP      -         sky launch -c myclus unsl...  
```